test/tests/unit/probe/router.test.ts

```diff
@@ -447,37 +447,73 @@ describe('probe router', () => {
 			expect(probes[0]!.location.country).to.equal('GB');
 		});
 
-		it('should return result sorted by priority of magic fields', async () => {
+		it('should return result sorted by priority of magic fields in case of partial match', async () => {
 			const sockets: Array<DeepPartial<Socket>> = [
-				await buildSocket('socket-3', { country: 'PL', normalizedCity: 'warsaw', normalizedNetwork: 'ultra development networks' }),
+				await buildSocket('socket-3', { country: 'CZ', normalizedCity: 'praga', normalizedNetwork: 'ultra development networks' }),
 				await buildSocket('socket-2', { country: 'RS', normalizedCity: 'belgrade', normalizedNetwork: 'belgrade networks' }),
 				await buildSocket('socket-1', { country: 'DE', normalizedCity: 'berlin', normalizedNetwork: 'berlin networks' }),
 			];
 			fetchSocketsMock.resolves(sockets as never);
 
 			const probes = await router.findMatchingProbes([
-				{ magic: 'de' },
+				{ magic: 'd' },
 			], 100);
 
 			expect(fetchSocketsMock.calledOnce).to.be.true;
 			expect(probes.length).to.equal(3);
 			expect(probes[0]!.location.country).to.equal('DE');
 			expect(probes[1]!.location.country).to.equal('RS');
+			expect(probes[2]!.location.country).to.equal('CZ');
+		});
+
+		it('should ignore matches on the lower levels after exact match', async () => {
+			const sockets: Array<DeepPartial<Socket>> = [
+				await buildSocket('socket-3', { country: 'VN', normalizedCity: 'hanoi', normalizedNetwork: 'ultra networks' }),
+				await buildSocket('socket-2', { country: 'RU', normalizedCity: 'vnukovo', normalizedNetwork: 'super networks' }),
+				await buildSocket('socket-1', { country: 'HU', normalizedCity: 'budapest', normalizedNetwork: '23VNet Kft' }),
+			];
+			fetchSocketsMock.resolves(sockets as never);
+
+			const probes = await router.findMatchingProbes([
+				{ magic: 'vn' },
+			], 100);
+
+			expect(fetchSocketsMock.calledOnce).to.be.true;
+			expect(probes.length).to.equal(1);
+			expect(probes[0]!.location.country).to.equal('VN');
+		});
```

Comment:
This matches what I described in my last comment, but now I'm thinking it would make more sense even here to include only the exact match (https://github.com/jsdelivr/globalping/issues/247#issuecomment-1339815722). The impact isn't big either way, though, as this is mostly likely going to happen only with networks.